### PR TITLE
add get_coin_id api to full_node

### PR DESCRIPTION
**Pull request by oncecsc**
calculate coin id by Coin(parent_coin_info,puzzle_hash,amout) in python is very convenient, but use other language such as node.js is complicated. so i write an api  to generate coin id for other language.

### CODE REVIEW

File: chia/rpc/full_node_rpc_api.py [hunk 1]
```diff
@@ -388,6 +389,22 @@ async def push_tx(self, request: Dict) -> Optional[Dict]:
...
+            raise ValueError("parent_coin_info not in request")
+        if "puzzle_hash" not in request:
+            raise ValueError("puzzle_hash not in request")
+        
+        if "amount" not in request:
+            raise ValueError("amount not in request")
+        if not isinstance(request["amount"], int):
+            raise ValueError("An integer amount or fee is required (too many decimals)")
+
+        return {"coin_id": Coin(hexstr_to_bytes(request["parent_coin_info"]), hexstr_to_bytes(request["puzzle_hash"]), int(request["amount"])).name()}
```
Comment (mariano54) [inline chia/rpc/full_node_rpc_api.py:407]: We should use uint64 not int here

**Review by mariano54 — COMMENTED** [review]
I need to think about it. I agree that it's kind of complicated, but we don't want to add tons of RPCs for every function. 
May I ask what use case you are working on?

File: chia/rpc/full_node_rpc_api.py (hunk 1 above)
Comment (oncecsc) [reply chia/rpc/full_node_rpc_api.py:407]: > I need to think about it. I agree that it's kind of complicated, but we don't want to add tons of RPCs for every function.
> May I ask what use case you are working on?

thanks

**Comment by oncecsc** [comment]
1. When scan block, the get_additions_and_removals return coins info  contents puzzle_hash, parent_coin_info and amount. I need coin id or transaction id to identify this coin. And the chiaexplorer.com can use coin id find this coin contents.

2. When push a transaction use `create_signed_transaction`, it returns a transaction contents transaction name, but we can't use it to query transaciton content in block chiaexplorer. so if i told someone "I send some xch to your address", he can't see the transaction until his wallet display the transaction, If got the coin id , just send the url like: https://www.chiaexplorer.com/blockchain/coin/0xe7306129b9bfcf65c21ae2710d26d907523708df1b2bb0a2a8061a9fa85bab31

**Comment by oncecsc** [comment]
read the chia-blockchain-gui and [chia-cilent](https://github.com/freddiecoleman/chia-client) code, but still not find help information. And in keybase  only be told to use `Coin(parent_coin_info, puzzle_hash, amount).name()` in python. If add `get_coin_id` RPC it will very convenient to get coin id with other languages and don't consider how to calculate it. I’m very curious why remove coin id from Coin object.

**Comment by mariano54** [comment]
The coinID can be obtained by hashing the fields. We could add this also to `get_additions_and_removals`.

**Comment by oncecsc** [comment]
> The coinID can be obtained by hashing the fields. We could add this also to `get_additions_and_removals`.
I try hashing the fields in js, but the result is different from Coin(parent_coin_info, puzzle_hash, amount).

**Pull request closed without merging**